pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -410,10 +413,10 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	// Sets the v1.NodeNetworkUnavailable condition to False.
 	ca.setNetworkCondition(node)
 
-	// Nodes in clusters WITHOUT multi-networking are expected to have only 1 network-interface
-	// with 1 alias IPv4 range and/or 1 IPv6 address. Multi-network clusters may have 1 interface
-	// with multiple aliases.
-	if len(instance.NetworkInterfaces) == 1 &&
+	// When multi-networking is disabled, the IPAM logic only considers the primary
```

Comment:
Let's further clarify here that (now) it is possible for a cluster to have nodes with multiple interfaces but NOT have multi-networking enabled. This means that IPAM should only be done for just the default network and the rest don't need any special handling.